src/main/java/seedu/address/logic/sortcomparators/BuyerComparator.java

```diff
@@ -0,0 +1,44 @@
+package seedu.address.logic.sortcomparators;
+
+import java.util.Comparator;
+import java.util.Optional;
+
+import seedu.address.model.buyer.Buyer;
+import seedu.address.model.buyer.Name;
+import seedu.address.model.pricerange.PriceRange;
+
+/**
+ * A comparator to compare two Buyers in the list.
+ */
+public class BuyerComparator implements Comparator<Buyer> {
+    private Optional<Comparator<Name>> nameComparator;
+    private Optional<Comparator<PriceRange>> priceRangeComparator;
+
+    /**
+     * Creates a BuyerComparator object.
+     */
+    public BuyerComparator(Comparator<Name> nameComparator, Comparator<PriceRange> priceRangeComparator) {
+        this.nameComparator = Optional.ofNullable(nameComparator);
+        this.priceRangeComparator = Optional.ofNullable(priceRangeComparator);
+    }
+
+    @Override
+    public int compare(Buyer firstBuyer, Buyer secondBuyer) {
+        if (nameComparator.isPresent()) {
+            return nameComparator.get().compare(firstBuyer.getName(), secondBuyer.getName());
+        }
+        // else statement
+        // remove "return 0" once implemented
+        return 0;
+    }
```

Comment:
if `nameComparator` & `priceRangeComparator` are both present, do we want to sort by name first or price first? 
Or do we want to restrict the users to sort by one condition